lib/asciidoctor/converter/factory.rb

```diff
@@ -44,14 +44,14 @@ class << self
         # Returns the default [Factory] singleton instance
         def default initialize_singleton = true
           return @__default__ || new unless initialize_singleton
-          # FIXME this assignment is not thread_safe, may need to use a ::Threadsafe helper here
+          # FIXME this assignment is not concurrent-ruby, may need to use a ::Threadsafe helper here
           @__default__ ||= begin
             # NOTE .to_s hides require from Opal
-            require 'thread_safe'.to_s unless defined? ::ThreadSafe
-            new ::ThreadSafe::Cache.new
+            require 'concurrent'.to_s unless defined? ::Concurrent
+            new ::Concurrent::Hash.new
           rescue ::LoadError
```

Comment:
> this is just for tests
`thread_safe` is not only for test. It is implemented.
Adding `gem 'thread_safe', '~> 0.3.0'` to `Gemfile` is meaningless, if we do not implement the conditional requiring `thread_safe` logic here.

Reply:
Aha! I see what you are saying. Let me think about it. We might be able to put stubs in place for Ruby 1.8.7 or we can just make this part of the 2.0.0 release.

Reply:
Okay, I will wait you :)